src/Source.h

```diff
@@ -47,6 +47,47 @@ class Source {
     return cur;
   }
 
+  static const char* skipBom(const char* begin) {
+    // Skip any unicode byte order marks
+    switch(begin[0]) {
+      // UTF-32BE
+      case '\x00':
+        if (begin[1] == '\x00' && begin[2] == '\xFE' && begin[3] == '\xFF') {
```

Comment:
Just remembered I faced this problem else where, and the answer is `boost::starts_with`. See e.g. `inComment()`:

```cpp
  static bool inComment(const char* cur, const char* end, const std::string& comment) {
    boost::iterator_range<const char*> haystack(cur, end);
    return boost::starts_with(haystack, comment);
  }
```